### PR TITLE
XOL-4975 Support for Symfony 3

### PR DESCRIPTION
**Pull request by anush**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /vendor/
+composer.lock
```
Comment (rushi) [inline .gitignore:2]: Why exclude composer.lock, don't we need it to pin down dependencies?

**Review by rushi — CHANGES_REQUESTED** [review]
Travis is failing

File: .gitignore (hunk 1 above)
Comment (anush) [reply .gitignore:2]: `composer.lock` is only used for the root repo. The lock file for dependencies are ignored and it doesn't make sense to keep it for projects that will only ever be used as dependencies.

**Comment by anush** [comment]
@rushi phpunit is failing on php 7.2 because for some reason it treats deprecation warnings are failures. I don't know how to resolve that, but the actual tests are passing.

**Pull request merged**